server/src/main/java/org/eclipse/openvsx/LocalRegistryService.java

```diff
@@ -1036,4 +1039,19 @@ public String getPublicKey(String publicId) {
 
         return keyPair.getPublicKeyText();
     }
+
+    @Override
+    public RegistryVersionJson getRegistryVersion() {
+        String propertiesPath = "version.properties";
+        Properties properties = new Properties();
+
+        try {
+            properties.load(new ClassPathResource(propertiesPath).getInputStream());
+            var registryVersion = new RegistryVersionJson();
+            registryVersion.version = properties.getProperty("version");
+            return registryVersion;
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to load version from " + propertiesPath, e);
```

Comment:
Other methods of the service throw `ErrorResultException`, should it be the same here?

Reply:
If I understand the code correctly, `ErrorResultException` seems to be used if the client side makes an argument error or uses an invalid token on a protected endpoint.

But you sure have a point: If the version cannot be read, we should return an 'unknown' version string instead of throwing runtime exceptions. I will address this in a follow-up commit.